### PR TITLE
Removed compilation step inside main_test.

### PR DESCRIPTION
**Pull request by jkohen**
Advantages:
* Speeds up the test slightly.
* One binary for the `main` function and test, which is how every other test works, and it ensures they are compiled identically.
* The test code is simpler.

### CODE REVIEW

**Comment by fabxc** [comment]
@brian-brazil could you take a look at this?

**Comment by brian-brazil** [comment]
I think this might be better controlled by a command line flag, which we can strip off before calling main()

File: cmd/prometheus/main_test.go [hunk 1]
```diff
@@ -31,37 +30,17 @@ import (
...
-	build := exec.Command("go", "build", "-o", promPath)
-	output, err := build.CombinedOutput()
-	if err != nil {
-		fmt.Printf("compilation error :%s \n", output)
-		os.Exit(1)
-	}
-
-	exitCode := m.Run()
-	os.Remove(promPath)
-	os.RemoveAll(promData)
```
Comment (roidelapluie) [inline cmd/prometheus/main_test.go]: That should still happen.

**Comment by brian-brazil** [comment]
Done by #6658

**Comment by jkohen** [comment]
Thanks all for the review.

**Pull request closed without merging**